src/bin/cargo/commands/help.rs

```diff
@@ -22,11 +22,21 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     let subcommand = args.get_one::<String>("COMMAND");
     if let Some(subcommand) = subcommand {
         if !try_help(config, subcommand)? {
-            crate::execute_external_subcommand(
-                config,
-                subcommand,
-                &[OsStr::new(subcommand), OsStr::new("--help")],
-            )?;
+            match check_builtin(&subcommand) {
+                Some(s) => {
+                    crate::execute_internal_subcommand(
+                        config,
+                        &[OsStr::new(s), OsStr::new("--help")],
+                    )?;
```

Comment:
Side note: a part of me would rather we do this without spawning but that would require calling into the top-level `clap::Command` with our own custom args while this ensures we have a unified experience.

Reply:
Did you mean like this?
```diff
diff --git a/src/bin/cargo/commands/help.rs b/src/bin/cargo/commands/help.rs
index 2839b931e..78fb3db59 100644
--- a/src/bin/cargo/commands/help.rs
+++ b/src/bin/cargo/commands/help.rs
@@ -22,12 +22,11 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     let subcommand = args.get_one::<String>("COMMAND");
     if let Some(subcommand) = subcommand {
         if !try_help(config, subcommand)? {
-            match check_builtin(&subcommand) {
+            match super::builtin_exec(&subcommand) {
                 Some(s) => {
-                    crate::execute_internal_subcommand(
-                        config,
-                        &[OsStr::new(s), OsStr::new("--help")],
-                    )?;
+                    let cmd = crate::cli::cli();
+                    let matches = cmd.get_matches_from(["cargo", subcommand, "--help"]);
+                    s(config, &matches)?;
                 }
                 None => {
                     crate::execute_external_subcommand(

```

Reply:
Something like that, yeah.